src/rgw/rgw_sal_motr.cc

```diff
@@ -1525,26 +1518,17 @@ MotrObject::~MotrObject() {
 int MotrObject::fetch_obj_entry_and_key(const DoutPrefixProvider* dpp, rgw_bucket_dir_entry& ent, std::string& bname, std::string& key, rgw_obj* target_obj)
 {
   int rc = this->get_bucket_dir_ent(dpp, ent);
+
   if (rc < 0) {
       ldpp_dout(dpp, 0) <<__func__<< ": ERROR: failed to get object entry. rc=" << rc << dendl;
       return rc;
-    }
-
-  read_bucket_info(dpp, bname, key, target_obj);
-
-  // getting key for version cases
-  if(ent.key.instance == "null")
```

Comment:
@andriytk  Could you please explain why do we need to remove this code?
This code is required to support versioning. Also seen simillar changes in this PR.

Reply:
The versioning works without this code, see explanation in the description above. Please let me know if it's not so, and what exactly does not work.

Reply:
We did some sanity testing for this patch and below are the list of scenario that might not work.

list-object-versions API:
- Null versions always listed at the top in the list-object-versions API response.
 
get-object API:
- get-object API might not work with --version-id null parameter for simple and multipart object
Expected o/p: object data for actual null version should be return.

head-object API:
- Head-object API might not work with --version-id null parameter for simple and multipart object.
Expected o/p: object metadata for actual null version should be return.
 
delete-object API:
- In suspended bucket delete-object api might not work with --version-id null parameter.
Expected o/p: It should delete null version.
 
- In case of suspended bucket, there might be a chances to have two null versions present in the bucket, one is null delete-marker and null version object.
Expected o/p: only one null entry should be present in the bucket - delete-marker/null version.

Reply:
Above scenarios are already listed in this doc -https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/1036157829/Test+scenarios+for+overall+versioning+feature

Reply:
Thanks. Please check the latest code - it should work now.